packages/pdf/src/RPA/PDF/keywords/document.py

```diff
@@ -0,0 +1,510 @@
+        :return: dictionary of pages and their texts
+
+        PDF needs to be parsed before text can be read.
+        """
+        self.switch_to_pdf_document(source_pdf)
+        if self.rpa_pdf_document is None:
+            self.ctx.convert()
+
+        if pages and not isinstance(pages, list):
+            pages = pages.split(",")
+        if pages is not None:
+            pages = list(map(int, pages))
+        pdf_text = {}
+        for idx, page in self.rpa_pdf_document.get_pages().items():
+            pdf_text[idx] = [] if details else ""
+            for _, item in page.get_textboxes().items():
+                if details:
+                    pdf_text[idx].append(item)
+                else:
+                    pdf_text[idx] += item.text
+        return pdf_text
+
+    @keyword
+    def extract_pages_from_pdf(
+        self, source_pdf: str = None, target_pdf: str = None, pages: Any = None
+    ) -> None:
+        """Extract pages from source PDF and save to target PDF document.
+
+        Page numbers start from 1.
+
+        :param source_pdf: filepath to the source pdf
+        :param target_pdf: filepath to the target pdf, stored by default
+            in `output_directory`
+        :param pages: page numbers to extract from PDF (numbers start from 0)
+            if None then extracts all pages
+        """
+        self.switch_to_pdf_document(source_pdf)
+        reader = PyPDF2.PdfFileReader(self.ctx.active_fileobject)
+        writer = PyPDF2.PdfFileWriter()
+
+        default_output = Path(self.output_directory / "extracted.pdf")
+        output_filepath = Path(target_pdf) if target_pdf else default_output
+
+        if pages and not isinstance(pages, list):
+            pages = pages.split(",")
+        elif pages is None:
+            pages = range(reader.getNumPages())
+        pages = list(map(int, pages))
+        for pagenum in pages:
+            writer.addPage(reader.getPage(int(pagenum) - 1))
+        with open(str(output_filepath), "wb") as f:
+            writer.write(f)
+
+    @keyword
+    def page_rotate(
+        self,
+        pages: int,
+        source_pdf: str = None,
+        target_pdf: str = None,
+        clockwise: bool = True,
+        angle: int = 90,
+    ) -> None:
+        """Rotate pages in source PDF document and save to target PDF document.
+
+        :param source_pdf: filepath to the source pdf
+        :param target_pdf: filepath to the target pdf, stored by default
+            to `output_directory`
+        :param pages: page numbers to extract from PDF (numbers start from 0)
+        :param clockwise: directorion that page will be rotated to, default True
+        :param angle: number of degrees to rotate, default 90
+        """
+        self.switch_to_pdf_document(source_pdf)
+        reader = PyPDF2.PdfFileReader(self.ctx.active_fileobject)
+        writer = PyPDF2.PdfFileWriter()
+
+        default_output = Path(self.output_directory / "rotated.pdf")
```

Comment:
Does the workflow make sense where each keyword call on a document stores it into a new file? Wouldn't you want to do a bunch of ops on the file in-memory and then as the final step store it on disk?